### PR TITLE
Add definitions for a few more standard OEP-2 keys

### PR DESCRIPTION
**Pull request by cpennington**
@nedbat, @efagin: This adds definitions for a few of the keys that we are using in the `openedx.yaml` files.

### CODE REVIEW

File: oeps/oep-0002.rst [hunk 1]
```diff
@@ -69,6 +70,32 @@ following fields:
...
+
+    .. This should be moved to a separate OEP
+
+    Whether or not this repository should be tracked for creation of OSPR tickets.
+
+``openedx-release``: dictionary (optional)
+
+    .. This should be moved to a separate OEP
+
+    This controls how this repo is managed during the edX Named Release process.
```
Comment (nedbat) [inline oeps/oep-0002.rst]: "during the Open edX release process".  We are moving away from "named release" as a term.

File: oeps/oep-0002.rst [hunk 2]
```diff
@@ -69,6 +70,32 @@ following fields:
     If an OEP isn't listed in the ``oeps`` dictionary, then it is assumed to be
     ``False``, unless the reporting tool can auto-detect accordance.
 
+``track-pulls``: boolean (optional)
+
+    .. This should be moved to a separate OEP
+
+    Whether or not this repository should be tracked for creation of OSPR tickets.
```
Comment (nedbat) [inline oeps/oep-0002.rst]: There's an irony to having this key any more.  When read from repos.yaml, it meant that we didn't have to look at every repo in the organization. Now that it's distributed, we have to look at every repo to see if this is set. Probably we should just check every repo no matter what.  This was just an optimization.

File: oeps/oep-0002.rst [hunk 3]
```diff
@@ -69,6 +70,32 @@ following fields:
...
+
+``openedx-release``: dictionary (optional)
+
+    .. This should be moved to a separate OEP
+
+    This controls how this repo is managed during the edX Named Release process.
+
+    Possible keys:
```
Comment (nedbat) [inline oeps/oep-0002.rst]: Suggested wording (plus another key):
```
    Possible keys:
    ``ref``
        The name of a branch in this repo. This is the branch that will be tagged when an Open edX release is made.
    ``requirements``
        The path to a requirements.txt-style file in this repo. This file will be consulted to find the commit to tag in dependent repos indicated by ``parent-repo`` keys.
    ``parent-repo``
        The name of another repo, one that installs this repo via a requirements file.  The parent repo's requirements file will indicate the commit to tag in this repo.
```

File: oeps/oep-0002.rst [hunk 4]
```diff
@@ -7,7 +7,7 @@ OEP-2: Repository Metadata
 +---------------+-------------------------------------------+
 | Title         | Repository Metadata                       |
 +---------------+-------------------------------------------+
-| Last-Modified | 2016-06-29                                |
+| Last-Modified | 2016-08-24                                |
```
Comment (nedbat) [inline oeps/oep-0002.rst]: There should be an explicit Change History section.

Comment (cpennington) [inline oeps/oep-0002.rst]: Good call, I'll add that to OEP-1.

File: oeps/oep-0002.rst (hunk 2 above)
Comment (cpennington) [inline oeps/oep-0002.rst]: Ah, I thought that was there because we didn't want the bot commenting on every repo.

File: oeps/oep-0002.rst (hunk 4 above)
Comment (cpennington) [inline oeps/oep-0002.rst]: Added in #19.

**Comment by nedbat** [comment]
:+1:

File: oeps/oep-0002.rst [hunk 5]
```diff
@@ -50,14 +50,15 @@ following fields:
 ``owner``: string
     The github user/team responsible for the repository.
 
-``nick``: string
-    A short-name for the repository used by reporting tools.
+``nick``: string (optional)
+    A short-name for the repository used by reporting tools created by the
+    edX Open Source team. Should be 3-4 characters.
```
Comment (benpatterson) [inline oeps/oep-0002.rst]: `Should be 3-4 characters` that is so helpful! thank you.

**Comment by benpatterson** [comment]
👍 from me. Thank you for putting in this update.

File: oeps/oep-0002.rst [hunk 6]
```diff
@@ -69,6 +70,39 @@ following fields:
...
+        The path to a requirements.txt-style file in this repo. This file will
+        be consulted to find the commit to tag in dependent repos indicated by
+        ``parent-repo`` keys.
+
+    ``parent-repo``
+        The name of another repo, one that installs this repo via a requirements
+        file.  The parent repo's requirements file will indicate the commit to
+        tag in this repo.
+
+``dead``: boolean (optional)
```
Comment (efagin) [inline oeps/oep-0002.rst]: Why can't we use a more standard (and less morbid) term like `obsolete` here? (You already use that in the description.)

File: oeps/oep-0002.rst (hunk 2 above)
Comment (efagin) [inline oeps/oep-0002.rst]: Agree. Sounds like we should rip this out and revisit if that use case ever emerges. Seems easy enough to bolt on later if we need it.

**Comment by efagin** [comment]
👍 this is technically merge-ready even though I had a few minor suggestions.

File: oeps/oep-0002.rst (hunk 6 above)
Comment (cpennington) [inline oeps/oep-0002.rst]: Yeah, I can do that. I'll update the existing repos that are marked w/ `dead`.

**Pull request merged**